addons/udes_stock_picking_batch/tests/test_stock_picking_batch.py

```diff
@@ -1,7 +1,231 @@
+        self.assertEqual(self.batch01.state, 'draft')
+
+        self.batch01.action_confirm()
+        self.assertEqual(self.batch01.state, 'waiting')
+
+        self.create_quant(self.apple.id, self.test_stock_location_01.id, 4,
+                          package_id=self.package_one.id)
+
+        self.picking01.action_assign()
+        self.assertEqual(self.batch01.state, 'ready')
+        # put in state 'in_progress'
+        self.assign_user()
+        self.assertEqual(self.batch01.state, 'in_progress')
+
+        # complete picks and check state done
+        self.complete_pick(self.picking01, call_done=False)
+        self.picking01._action_done()
+        self.assertEqual(self.batch01.state, 'done')
+
+    def test_check_computing_cancel(self):
+        """ Test done with cancel to check computation"""
+        self.draft_to_ready()
+        self.assign_user()
+
+        # Cancel the pick and confirm we reach state done
+        self.picking01.action_cancel()
+        self.assertEqual(self.batch01.state, 'done')
+
+    def test_check_computing_cancel(self):
+        """ Test done with cancel to check computation"""
+        self.draft_to_ready()
+        self.assign_user()
+
+        # set batch_id to False and check state 'done'
+        self.picking01.batch_id = False
+        self.assertEqual(self.batch01.state, 'done')
+
+    def test_computing_ready_picking_to_batch(self):
+        """ Test done with ready picking to check computation"""
+        self.create_quant(self.apple.id, self.test_stock_location_01.id, 4,
+                          package_id=self.package_one.id)
+        # assign picking before adding to batch
+        # called when no pick.state change occurs
+        self.picking01.action_assign()
+        self.assertEqual(self.batch01.state, 'draft')
+
+        # confirm picking and check compute_state is run
+        self.batch01.action_confirm()
+        self.assertEqual(self.batch01.state, 'ready')
+
+    def test_computing_partial_assignment(self):
+        """ Test done with partially complete pickings to check computation"""
+        self.draft_to_ready()
+        self.assign_user()
+
+        # Create second quant and assign picking, confirm 'in_progress' state
+        self.create_quant(self.apple.id, self.test_stock_location_01.id, 4,
+                          package_id=self.package_two.id)
+
+        self.picking02.action_assign()
+        self.picking02.batch_id = self.batch01
+        self.assertEqual(self.batch01.state, 'in_progress')
+
+        # complete pick1 and check state 'in_progress', forcibly compute_state
+        self.complete_pick(self.picking01, call_done=False)
+        self.picking01._action_done()
+        self.assertEqual(self.batch01.state, 'in_progress')
+
+        # complete pick2 and check state 'done', forcibly compute_state
+        self.complete_pick(self.picking02, call_done=False)
+        self.picking02._action_done()
+        self.assertEqual(self.batch01.state, 'done')
```

Comment:
Missing newline and also using single quotes, can you black format all these files please?

Reply:
Done